### PR TITLE
Deprecate isSingleRefUnevaluated

### PR DESCRIPTION
**Pull request by fjeremic**
See eclipse/omr#5648 for discussion. It was decided during the OMR
architecture meeting that this API is not very clear and that inlining
it's uses is more understandable as it is very explicit as to what is
being checked.

Signed-off-by: Filip Jeremic <fjeremic@ca.ibm.com>

### CODE REVIEW

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8 depends eclipse/omr#5797

**Comment by fjeremic** [comment]
@0xdaryl do you want to be the committer for this one?

**Comment by fjeremic** [comment]
Although it doesn't really depend on OMR I'm tagging it as such here because this actually needs to go in first. Otherwise OMR-Acceptance will be blocked.

**Comment by 0xdaryl** [comment]
I suggest you don't tag it with "depends:omr" because that's not what the label has come to mean (that an OMR change must merge first before the OpenJ9 one).  I think you just need to keep the OMR change in WIP/draft until the OpenJ9 change merges.

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**
